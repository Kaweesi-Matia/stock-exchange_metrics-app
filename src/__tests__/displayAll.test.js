/* eslint-disable  */
import React from 'react';

import { render, screen } from '@testing-library/react';
import * as router from 'react-router';
import { Provider } from 'react-redux';
import store from '../redux/Store';
import DisplayAll from '../components/displayAll';
import StockExchangeData from '../components/stockExchangeData';




describe('expected to get a correct output', () => {
    const Navigation = jest.fn();  
    beforeEach(() => {
      jest.spyOn(router, 'useNavigate').mockImplementation(() => Navigation);
    });
    
    test('it must render the page for particulars', () => {
      render(<Provider store={store}><StockExchangeData /></Provider>);
      expect(screen.findAllByText('Price change(usd):')).toMatchSnapshot();
    });
    test('must render the home page', () => {
        render(<Provider store={store}><DisplayAll /></Provider>);
        expect(screen.findAllByText(' Crypto currency market data')).toMatchSnapshot();
      });
   

   
  });
  