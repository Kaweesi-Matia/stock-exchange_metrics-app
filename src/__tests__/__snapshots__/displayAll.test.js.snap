// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ensure app renders as expected renders the Details page 1`] = `Promise {}`;

exports[`Ensure app renders as expected renders the home page 1`] = `Promise {}`;

exports[`expected to get a correct output  1`] = `Promise {}`;

exports[`expected to get a correct output it must render 1`] = `Promise {}`;

exports[`expected to get a correct output it must render the page for particu 1`] = `Promise {}`;

exports[`expected to get a correct output it must render the page for particulars 1`] = `Promise {}`;

exports[`expected to get a correct output must render the home p 1`] = `Promise {}`;

exports[`expected to get a correct output must render the home page 1`] = `Promise {}`;

exports[`expected to get a correct output renders the Details page 1`] = `Promise {}`;

exports[`expected to get a correct output renders the home page 1`] = `Promise {}`;
